rust/rope/src/delta.rs

```diff
@@ -22,10 +22,16 @@ use subset::{Subset, SubsetBuilder};
 use std::cmp::min;
 
 enum DeltaElement<N: NodeInfo> {
+    /// Represents the text in a range of the base document. Includes beginning, excludes end.
     Copy(usize, usize),  // note: for now, we lose open/closed info at interval endpoints
     Insert(Node<N>),
 }
 
+/// Represents an entire new document as a sequence of sections copied from
```

Comment:
I find this a little confusing. I suppose it is a representation of a document, but I find it easier to think of it as a representation of a _transformation_ from one document (or document state) to another. 

This PR is a welcome opportunity for me to clarify my own understanding of a bunch of this code. 

How I would write this:

"A representation of a transformation from one document state to another. 

This transformation is composed of a series of `Copy` and `Insert` operations. `Copy` operations indicate unmodified ranges of the existing document, and `Insert` operations indicate additions. 

Note: The deletion of some part of the document, in this schema, is represented as a set of `Copy` operations over the ranges to be kept.

For example, the transformation from "hello" to "hippo" can be represented as:
`[Copy(0, 1), Insert("ipp"), Copy(4,5)]`"



(sorry, I got a little carried away)